sdcm/remote.py

```diff
@@ -53,10 +53,28 @@ class SSHConnectTimeoutError(Exception):
     """
 
 
-NETWORK_EXCEPTIONS = (NoValidConnectionsError, SSHException, SSHConnectTimeoutError, EOFError,
+NETWORK_EXCEPTIONS = (NoValidConnectionsError, SSHException, SSHConnectTimeoutError, EOFError, AuthenticationException,
                       ConnectionResetError, ConnectionAbortedError, ConnectionError, ConnectionRefusedError)
 
 
+class RetriableNetworkException(Exception):
+    """
+        SSH protocol exception that can be safely retried
+    """
+
+
+def is_exception_retriable(err_str):
+    """Check that exception can be safely retried"""
+    exceptions = ("Authentication timeout", "Error reading SSH protocol banner", "Timeout opening channel",
+                  "Unable to open channel", "Key-exchange timed out waiting for key negotiation",
+                  "ssh_exchange_identification: Connection closed by remote host",
+                  )
+    for exception_str in exceptions:
+        if exception_str in err_str:
+            return True
+    return False
+
+
```

Comment:
perhaps an idea for improvement, generalize this to all exceptions and errors, could be writing the executed requested command and its timestamp to a file on the remote node after the command was executed. this gives more certainty if the command is already executed or can be retried for many more types of connection errors.

Reply:
As I said in one of my comments, verifying the reliable protocol will not give any advantage. If there is a network issue you need to know about it and resolve the root cause.